### PR TITLE
Support K2

### LINKED ISSUE #118: Support K2
It's available now as a preview, and will become the default for Kotlin 2.0. ZacSweers/redacted-compiler-plugin#86 might be a good reference.

### PR DESCRIPTION
**Pull request by drewhamilton**
Closes #118. Poko now works for Kotlin language version 2.0 (and continues to work for 1.9).

### CODE REVIEW

File: poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/PokoCompilerPluginRegistrar.kt [hunk 1]
```diff
@@ -16,8 +16,7 @@ import org.jetbrains.kotlin.name.ClassId
 @AutoService(CompilerPluginRegistrar::class)
 public class PokoCompilerPluginRegistrar : CompilerPluginRegistrar() {
 
-    // TODO: Support K2
-    override val supportsK2: Boolean = false
+    override val supportsK2: Boolean = true
```
Comment (JakeWharton) [inline poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/PokoCompilerPluginRegistrar.kt:19]:
```suggestion
    override val supportsK2: Boolean get() = true
```
😀 save a field!

**Review by JakeWharton — APPROVED** [review]
Well that was easy...

**Comment by drewhamilton** [comment]
Spoke too soon! JVM works but native fails.

```
> Task :poko-tests:compileKotlinIosArm64K2 FAILED
w: Language version 2.0 is experimental, there are no backwards compatibility guarantees for new language and library features
e: Compilation failed: Collection contains more than one matching element.

 * Source files: Super.kt, AnyArrayHolder.kt, Complex.kt, ExplicitDeclarations.kt, Nested.kt, Simple.kt, SuperclassDeclarations.kt, AnyArrayHolder.kt, ArrayHolder.kt, Complex.kt, ComplexGenericArrayHolder.kt, ExplicitDeclarations.kt, GenericArrayHolder.kt, Nested.kt, Simple.kt, SimpleWithExtraParam.kt, SuperclassDeclarations.kt
 * Compiler version: 1.9.10
 * Output kind: LIBRARY

e: java.lang.IllegalArgumentException: Collection contains more than one matching element.
	at org.jetbrains.kotlin.fir.backend.IrBuiltInsOverFir$extensionToString$2.invoke(IrBuiltInsOverFir.kt:1169)
	at org.jetbrains.kotlin.fir.backend.IrBuiltInsOverFir$extensionToString$2.invoke(IrBuiltInsOverFir.kt:524)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at org.jetbrains.kotlin.fir.backend.IrBuiltInsOverFir.getExtensionToString(IrBuiltInsOverFir.kt:524)
	at dev.drewhamilton.poko.ir.ToStringGenerationKt.irRuntimeArrayContentDeepToString(toStringGeneration.kt:156)
	at dev.drewhamilton.poko.ir.ToStringGenerationKt.generateToStringMethodBody(toStringGeneration.kt:70)
	at dev.drewhamilton.poko.ir.PokoMembersTransformer.visitFunctionNew(PokoMembersTransformer.kt:60)
	at ...
```

Will take a closer look tomorrow.

**Comment by drewhamilton** [comment]
Added another workaround and filed [KT-61627](https://youtrack.jetbrains.com/issue/KT-61627).

**Pull request merged**